test/lib/mayaUsd/nodes/testProxyShapeBase.py

```diff
@@ -496,8 +511,24 @@ def verifyPrim():
             cmds.file(save=True, force=True, type='mayaAscii')
 
             # Verify that the prim is still inactive in the target layer.
+            self._verifyPrim(False)
+            self._verifySubLayer()
+
+            # Reload the file and verify again.
+            cmds.file(new=True, force=True)
+            cmds.file(tempMayaFile, force=True, open=True)
 
-            verifyPrim()
+            self._verifyPrim(False)
+            self._verifySubLayer()
+
+            # Change shared status and verify again. Changing the shared flag recomputes
+            # the stage and its layer, which is what we really want to test here. So we
+            # change an input attribute that we know will recompute the layers.
+            cmds.setAttr('{}.{}'.format(proxyShapePath,"shareStage"), False)
+    
```

Comment:
These extra checks, after reload + modify stage are what were triggering the layer lost bug.